### PR TITLE
pubsublite(consumer): Add minimal logging

### PR DESCRIPTION
**Pull request by marclop**
Adds a bit more logging to the PubSub Lite consumer. This should make it easier to spot any processing errors and re-processing attempts.

Added while working on fixing #116

### CODE REVIEW

File: pubsublite/consumer.go [hunk 1]
```diff
@@ -201,7 +201,10 @@ func NewConsumer(ctx context.Context, cfg ConsumerConfig) (*Consumer, error) {
 func (c *Consumer) Close() error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
-	c.stopSubscriber()
+	if c.stopSubscriber != nil {
```
Comment (marclop) [inline pubsublite/consumer.go:204]: If the consumer hadn't been started, close would panic.

File: pubsublite/consumer.go [hunk 2]
```diff
@@ -283,10 +287,12 @@ func (c *consumer) processMessage(ctx context.Context, msg *pubsub.Message) {
 					attempt += a.(int)
 				}
 				if attempt > 2 {
+					c.logger.Info("re-processing limit exceeded, discarding event", zap.Int("attempt", attempt))
```
Comment (marclop) [inline pubsublite/consumer.go:290]: Should this be a warn?

Comment (axw) [reply pubsublite/consumer.go:290]: I think so

File: pubsublite/consumer.go [hunk 3]
```diff
@@ -228,6 +231,7 @@ func (c *Consumer) Run(ctx context.Context) error {
 				))
 				// Keep attempting to receive until a fatal error is received.
 				if errors.Is(err, pscompat.ErrBackendUnavailable) {
+					c.cfg.Logger.Info("transient error found, re-starting consumer...", zap.Error(err))
```
Comment (dmathieu) [inline pubsublite/consumer.go:234]: We should have a span here. So we could also fetch it from context and add an event to it (so it's easier to search and match to a specific event)

Comment (marclop) [reply pubsublite/consumer.go:234]: it seems strange to me to have a span here. `Receive()` may block for hours or days if no transient or fatal errors occur. Is this what you had in mind?

Comment (dmathieu) [reply pubsublite/consumer.go:234]: Ah no, right. Somehow, I thought it returned the received object. Yeah, in that case, it makes sense. There shouldn't be any spans (we could create one to pass the log message, but that'd just be a different way to do logging).

**Pull request merged**